net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,263 @@
+  procd_set_param stderr "$std_err"
+  procd_close_instance
+  (
+    # Use flock to acquire an exclusive lock
+    local LOCK_FILE=/var/lock/tailscale.lock
+    exec 9> "$LOCK_FILE"
+    flock -xn 9 || {
+        echo "Failed to acquire lock on $LOCK_FILE"
+        exit 1
+    }
+
+    # Wait for Tailscale IPv4 address to be available
+    count=0
+    while [ -z "$(ifconfig | grep 'tailscale' | awk '{print $1}')" ] || [ -z "$(tailscale ip -4)" ]
+    do
+      sleep 2
+      let count++
+      [ "${count}" -ge 5 ] && exit 19
+    done
+
+    # Configure Tailscale MagicDNS
+    if [ "$accept_dns" = "1" ]; then
+      MagicDNSSuffix=$(tailscale status --json | awk -F'"' '/"MagicDNSSuffix"/ {last=$(NF-1)} END {print last}')
+      target_address="/$MagicDNSSuffix/100.100.100.100"
+      index=$(uci show dhcp | grep 'dhcp.@dnsmasq\[[0-9]\+\]=dnsmasq' | grep -o '[0-9]\+')
+      for i in $index; do
+        uci get dhcp.@dnsmasq[$i].address 2>/dev/null | grep -qxF "$target_address" || uci add_list dhcp.@dnsmasq[$i].address="$target_address"
+      done
+    fi
+
+    # Configure network interface for Tailscale
+    ts0=$(ifconfig | grep 'tailscale' | awk '{print $1}')
+    if [ -z "$(uci -q get network.tailscale)" ]; then
+      uci set network.tailscale='interface'
+      if [ "$ts0" = *$'\n'* ]; then
+        uci set network.ts_lan='device'
+        uci set network.ts_lan.type='bridge'
+        uci set network.ts_lan.name='ts-lan'
+        for port in "${ts0}"; do
+          uci add_list network.ts_lan.ports=$port
+        done
+        uci set network.tailscale.proto='none'
+        uci set network.tailscale.device='ts-lan'
+      else
+        ts_ip=$(tailscale ip -4)
+        uci set network.tailscale.proto='static'
+        uci set network.tailscale.ipaddr=$ts_ip
+        uci set network.tailscale.netmask='255.0.0.0'
+        uci set network.tailscale.device=$ts0
+      fi
+    fi
+
+    # Find the LAN to WAN forwarding rule index
+    index=$(uci show firewall | grep "firewall.@forwarding\[[0-9]\+\]\.src='lan'" -B 1 -A 1 | grep "firewall.@forwarding\[[0-9]\+\]\.dest='wan'" | grep -o '[0-9]\+')
+    # Configure firewall forwarding rules based on exit_node
+    if [ -n "$exit_node" ]; then
+      uci set firewall.@defaults[0].forward='REJECT'
+      [ -n $index ] && uci set firewall.@forwarding[$index].enabled='0'
+    fi
+
+    # Configure subnet routes for site to site
+    config_get subnet_routes $cfg 'subnet_routes'
+    if [ -n "$subnet_routes" ]; then
+      i=1
+      ts_ip=$(tailscale ip -4)
+      for route in $subnet_routes; do
+        uci set network.ts_subnet$i='route'
+        uci set network.ts_subnet$i.interface='tailscale'
+        uci set network.ts_subnet$i.target=$route
+        uci set network.ts_subnet$i.gateway=$ts_ip
+        let i++
+      done
+    fi
+
+    # Create firewall zone and forwarding rules for Tailscale
+    config_get access $cfg 'access'
+    if [ -n "$access" ]; then
+      uci batch <<-EOF
```

Comment:
If you want to use the tab suppression syntax `<<-` you must use tabs for indentation in heredocs. Which if you do will lead to mixing tabs and spaces within the same file, so you might as well consider using tabs everywhere.